### PR TITLE
TASK SIO-2631 Compute Resource list, Compute Resource Storage list, Storage delete

### PR DESCRIPTION
**Pull request by oneumyvakin**

### CODE REVIEW

File: computeResource.go [hunk 1]
```diff
@@ -152,6 +152,13 @@ type computerResourceResponse struct {
 	Data ComputeResource `json:"data"`
 }
 
+func (s *ComputeResourcesService) List(ctx context.Context) ([]ComputeResource, error) {
+	resp := struct {
+		Data []ComputeResource `json:"data"`
+	}{}
+	return resp.Data, s.client.list(ctx, "compute_resources", &resp)
```
Comment (dshemin) [inline computeResource.go:159]: I think we should return an paginated response here like we do in other methods.

File: computeResourcesStorage.go [hunk 2]
```diff
@@ -20,3 +20,10 @@ func (s *ComputeResourcesService) StorageCreate(
 	var resp storageResponse
 	return resp.Data, s.client.create(ctx, fmt.Sprintf("compute_resources/%d/storages", id), data, &resp)
 }
+
+func (s *ComputeResourcesService) StorageList(ctx context.Context, id int) ([]Storage, error) {
+	resp := struct {
+		Data []Storage `json:"data"`
+	}{}
+	return resp.Data, s.client.get(ctx, fmt.Sprintf("compute_resources/%d/storages", id), &resp)
```
Comment (dshemin) [inline computeResourcesStorage.go:28]: Here we should return an paginated response too.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: computeResourcesStorage.go (hunk 2 above)
Comment (oneumyvakin) [reply computeResourcesStorage.go:28]: In SolusIO response of compute_resources/%d/storages is not paginated.

**Review by kryptamine — APPROVED** [review]

File: computeResource.go [hunk 3]
```diff
@@ -152,6 +152,19 @@ type computerResourceResponse struct {
 	Data ComputeResource `json:"data"`
 }
 
+func (s *ComputeResourcesService) List(ctx context.Context) ([]ComputeResource, error) {
```
Comment (dshemin) [inline computeResource.go:155]: I mean that we should return paginated response and let users iterate other it.

**Review by dshemin — CHANGES_REQUESTED** [review]

File: computeResource.go (hunk 3 above)
Comment (oneumyvakin) [reply computeResource.go]: Got you.

**Review by dshemin — APPROVED** [review]

**Comment by dshemin** [comment]
LGTM!

**Pull request merged**